aggregate/billing_functions/seqr.py

```diff
@@ -0,0 +1,593 @@
+                batch_resource, usage
+            )
+
+            for dataset, fraction in prop_map.items():
+                # Distribute the remaining cost across all datasets proportionately
+                cost = gross_cost * fraction
+
+                entries.append(
+                    utils.get_entry(
+                        key='-'.join(
+                            [
+                                SERVICE_ID,
+                                'distributed',
+                                dataset,
+                                batch_resource,
+                                'batch',
+                                str(batch_id),
+                                'job',
+                                str(job_id),
+                            ]
+                        ),
+                        topic=dataset,
+                        service_id=SERVICE_ID,
+                        description='Seqr compute (distributed)',
+                        cost=cost,
+                        currency_conversion_rate=currency_conversion_rate,
+                        usage=round(usage * fraction),
+                        batch_resource=batch_resource,
+                        start_time=start_time,
+                        end_time=end_time,
+                        labels={
+                            **labels,
+                            'dataset': dataset,
+                            'fraction': round(100 * fraction) / 100,
+                        },
+                    )
+                )
+
+    entries.extend(utils.get_hail_credits(entries))
+
+    return entries
+
+
+def migrate_entries_from_bq(
+    start, end, prop_map: ProportionateMapType, dry_run=False
+) -> int:
+    """
+    Migrate entries from BQ to GCP, using the given proportionate maps
+    """
+    # pylint: disable=too-many-branches
+    _query = f"""
+        SELECT * FROM `{GCP_BILLING_BQ_TABLE}`
+        WHERE export_time >= @start
+            AND export_time <= @end
+            AND project.id = @project
+        ORDER BY usage_start_time
+    """
+
+    job_config = bq.QueryJobConfig(
+        query_parameters=[
+            bq.ScalarQueryParameter('start', 'STRING', str(start)),
+            bq.ScalarQueryParameter('end', 'STRING', str(end)),
+            bq.ScalarQueryParameter('project', 'STRING', 'seqr-308602'),
+        ]
+    )
+
+    df = (
+        utils.bigquery_client.query(_query, job_config=job_config)
+        .result()
+        .to_dataframe()
+    )
+    json_obj = json.loads(df.to_json(orient='records'))
+
+    entries_by_id = {}
+    result = 0
+    param_map, current_date = None, None
+    for obj in json_obj:
+
+        if obj['cost'] == 0:
+            # come on now
+            continue
+        usage_start_time = datetime.fromtimestamp(int(obj['usage_start_time'] / 1000))
+        del obj['billing_account_id']
+        del obj['project']
+        if current_date is None or usage_start_time > current_date:
```

Comment:
This is a small optimisation to avoid calling get_ratios over and over again, and noting that it's sorted by date

Reply:
Eep, this isn't implemented correctly either